### DISCUSSION TITLE
[@types/cheerio] cheerio type definitions are woefully out of date, possibly by design?

### DISCUSSION THREAD

**Discussion by josheverett** [DefinitelyTyped/DefinitelyTyped#61157] [Issues with a @types package] [answered]
Line 1 in types/cheerio/index.d.ts:

`// Type definitions for Cheerio v0.22.0`

This older version of cheerio was published 6 years (!) ago and is no longer maintained:

https://github.com/cheeriojs/cheerio/tree/aa90399c9c02f12432bfff97b8f1c7d8ece7c307

As is, these type definitions are mostly-but-not-entirely compatible with newer cheerio versions. Notably, the `cheerio.load` is incompatible with newer overloads.

At the time of writing, the most recent version of cheerio is [v1.0.0-rc.12](https://github.com/cheeriojs/cheerio/releases/tag/v1.0.0-rc.12).

I realize that the "current" version is an RC, so perhaps it's technically correct that these type definitions are so out of date. But in practice, this is a rather silly inconvenience.

[a comment by an automated account is not shown]

**Comment by AzSiAz** [comment, marked as answer]
Last time I checked cheerio decided to bundle it's own type definition, you might want to check if that's still the case
